### DISCUSSION TITLE
Should wire_gen.go be checked into version control?

### DISCUSSION THREAD

**Discussion by naftulikay** [google/wire#325] [Q&A]
Since `wire_gen.go` files are generated and not user-authored, should they be checked-in to version control? They seem to be idempotent, they don't change unless injectors change, but I'm not sure on whether to check them in or not as they are generated artifacts.

Additionally, what is the strategy for building a project in CI/CD such as GitHub Actions? Do I need to install the `wire` command in my build environment, or does `go generate` cover things? I'm going to start testing this out on my own to see what the different possibilities are, but I'd like to hear best practices from developers and users.

<sub>search tags: git, vcs, version control</sub>

**Comment by cglotr** [comment]
I think it's better to check in the generated code since it's a part of the code. Also, there are a few advantages I think:
- Able to identify any diff from new code generated by `wire` if we are using `wire@latest`
- Simplify & speedup CI/CD since no need to do `go install github.com/google/wire/cmd/wire@latest` step & run `wire`
- Can optionally write tests to Wire generated code

For CI/CD, if you don't check in the code then you have to install & run Wire in the build step:

```
go install github.com/google/wire/cmd/wire@latest
wire
```
